src/App.tsx

```diff
@@ -15,35 +16,122 @@
   'Garlic',
 ];
 
+enum SortType {
+  DEFAULT,
+  ALPHABETICALLY,
+  LENGTH,
+}
+
+interface SortingConditionals {
+  sortingType: SortType;
+  isReversed: boolean;
+}
+
+const sortingGoodsFunction = (
+  goodsArray: string[],
+  { sortingType, isReversed }: SortingConditionals,
+) => {
+  const goodsArrayCopy = [...goodsArray];
+
+  switch (sortingType) {
+    case SortType.ALPHABETICALLY:
+      goodsArrayCopy.sort((good1, good2) => good1.localeCompare(good2));
+      break;
+
+    case SortType.LENGTH:
+      goodsArrayCopy.sort((good1, good2) => good1.length - good2.length);
+      break;
+
+    default:
+      break;
+  }
+
+  if (isReversed === true) {
+    return goodsArrayCopy.reverse();
+  } else {
+    return goodsArrayCopy;
+  }
+};
+
 export const App: React.FC = () => {
+  const [sortingType, setSortingType] = useState<SortType>(SortType.DEFAULT);
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+
+  const handleReset = () => {
+    setSortingType(SortType.DEFAULT);
+    setIsReversed(false);
+  };
+
+  const handleReverse = () => {
+    if (isReversed === false) {
+      setIsReversed(true);
+    }
+
+    if (isReversed === true) {
+      setIsReversed(false);
+    }
```

Comment:
The `handleReverse` function could be simplified by directly toggling the `isReversed` state instead of using conditional checks. This would make the code cleaner and more concise. Consider using `setIsReversed(!isReversed);`.